job-service-db/src/main/resources/procedures/getJob.sql

```diff
@@ -53,9 +57,12 @@ BEGIN
            job.status,
            job.percentage_complete,
            job.failure_details,
-           CAST('WORKER' AS CHAR(6)) AS actionType
+           CAST('WORKER' AS CHAR(6)) AS actionType,
+           jtd.job_id IS NULL AS can_be_progressed
     FROM job
-    WHERE job.job_id = in_job_id;
+    LEFT JOIN job_task_data AS jtd ON jtd.partition = job.partition AND jtd.job_id = job.job_id
```

Comment:
You're obviously introducing this just for `can_be_progressed`.  Let's make the lookup for that a separate function, rather than including it here.